ballerina/Module.md

```diff
@@ -1,17 +1,81 @@
 ## Overview
 
-[//]: # (TODO: Add overview mentioning the purpose of the module, supported REST API versions, and other high-level details.)
+[OpenAI](https://openai.com/), an AI research organization focused on creating friendly AI for humanity, offers the [OpenAI API](https://platform.openai.com/docs/api-reference/introduction) to access its powerful AI models for tasks like natural language processing and image generation.
+
+The `ballarinax/openai.audio` package offers APIs to connect and interact with [Audio related endpoints of OpenAI API v1](https://platform.openai.com/docs/api-reference/audio/create) providing access to various models developed by OpenAI for audio-related tasks.
 
 ## Setup guide
 
-[//]: # (TODO: Add detailed steps to obtain credentials and configure the module.)
+To use the OpenAI Connector, you must have access to the OpenAI API through a [OpenAI Platform account](https://platform.openai.com) and a project under it. If you do not have a OpenAI Platform account, you can sign up for one [here](https://platform.openai.com/signup).
+
+#### Create a OpenAI API Key
+
+1. Open the [OpenAI Platform Dashboard](https://platform.openai.com).
+
+2. Navigate to Dashboard -> API keys
+![navigate-api-key-dashboard](https://github.com/e19166/module-ballerinax-openai.audio/blob/main/docs/setup/resources/navigate-api-key-dashboard.png)
+
+3. Click on the "Create new secret key" button
+![api-key-dashboard](https://github.com/e19166/module-ballerinax-openai.audio/blob/main/docs/setup/resources/api-key-dashboard.png)
+
+4. Fill the details and click on Create secret key
+![create-new-secret-key](https://github.com/e19166/module-ballerinax-openai.audio/blob/main/docs/setup/resources/create-new-secret-key.png)
+
+5.  Store the API key securely to use in your application
+![saved-key](https://github.com/e19166/module-ballerinax-openai.audio/blob/main/docs/setup/resources/saved-key.png)
 
 ## Quickstart
```

Comment:
For the quickstart, lets stick to the exact same structure used in https://github.com/ballerina-platform/module-ballerinax-openai.images/blob/main/ballerina/Module.md#quickstart